src/SCRIPTS/BF/ui.lua

```diff
@@ -354,9 +354,14 @@ function run_ui(event)
             incValue(-1)
         end
     end
+    local nextPage = nil
     while Page == nil do
         Page = assert(loadScript(radio.templateHome .. PageFiles[currentPage]))()
-        if Page.requiredVersion and Page.requiredVersion > apiVersion then
+        if Page.requiredVersion and apiVersion and Page.requiredVersion > apiVersion and currentPage ~= nextPage then
```

Comment:
`apiVersion` will always evaluate to true of it is not nil.

> \> x = 0
> \> if x then print(1) end
1

Given that `apiVersion` is initialized as 0 and is never nil, the boolean check for this variable will always be true
https://github.com/betaflight/betaflight-tx-lua-scripts/blob/62d7d4af61811e4bb752d401c54cc96b2bea3fdc/src/SCRIPTS/TELEMETRY/bf.lua#L3

Perhaps nesting `currentPage ~= nextPage` within this if statement would permit a way to gracefully exit when they are equivalent.

Reply:
I have added exiting if no valid pages are found.